nservicebus/hosting/critical-errors.md

```diff
@@ -44,14 +44,29 @@ Next define what action to take when this scenario occurs:
 snippet:CustomHostErrorHandlingAction
 
 
-## When to override the default action
+## When to override the default critical error action
 
-The default action should be overridden whenever that default does not meet the specific hosting requirements. For example
+The default behavior is to stop the endpoint and not the process. It is recommended to exit the process when a critical error occurs. Not overriding the critical error callback will result in the endpoint instance to stop processing messages without the ability to recover from this state.
 
- * If using [NServiceBus Host](/nservicebus/hosting/nservicebus-host), and wish to take a custom action before the endpoint process is killed.
- * If self hosting the process can be shut down the process via [Environment.FailFast](https://msdn.microsoft.com/en-us/library/dd289240.aspx) and re-start the process once the root cause has been diagnosed.
+NOTE: If not killing the process and just disposing the bus, be aware that any `Send` operations will result in [ObjectDisposedException](https://msdn.microsoft.com/en-us/library/system.objectdisposedexception.aspx) being thrown.
+
+Consider the following when implementing a critical error callback:
+
+- If calling any code wrap it in a `try..finally`, making sure the process will invoke exit code in the `finally` like `Environment.FailFast` if any code fails.
+- Flush any loggers used which makes sure any unwritten log state is written/pushed to its target(s) making sure the tail of the log is not lost as [Environment.FailFast](https://msdn.microsoft.com/en-us/library/dd289240.aspx) will immediately exit the process.
+ - Flush appenders for [NLog](http://nlog-project.org/documentation/v4.3.0/html/M_NLog_LogManager_Shutdown.htm) or [log4net](https://logging.apache.org/log4net/log4net-1.2.11/release/sdk/log4net.LogManager.Shutdown.html) state by calling `LogManager.Shutdown();`.
+- Flush any state/caches used if such data structures would be written/persisted at a certain interval or regular graceful shutdown.
+- It is easiest to call [Environment.FailFast](https://msdn.microsoft.com/en-us/library/dd289240.aspx) or [Environment.Exit](https://msdn.microsoft.com/en-us/library/system.environment.exit(v=vs.110).aspx).
+
+
+Rely on the environment hosting the endpoint process for it to be automatically restarted.
```

Comment:
> Rely on the environment hosting the endpoint process for it to be automatically restarted

What are we trying to say here? That we should try to avoid having to restart processes manually, when they crash? What about other hosting options (other than IIS, Windows Service with Windows Service Recovery enabled), should we in other cases manually restart it?